src/kagglehub/models_helpers.py

```diff
@@ -9,18 +9,32 @@
 logger = logging.getLogger(__name__)
 
 
+class Directory:
+    name: str
+    files: List[str]
+    directories: List["Directory"]
+
+
+FileStructure = List[Directory]
```

Comment:
Instead of using FileStructure, you could simply pass a "Directory" which would essentially be the "root" directory along with the files & subdirectories.